packages/router/src/state_manager.ts

```diff
@@ -98,6 +106,15 @@ export class StateManager {
     };
   }
 
+  nonRouterCurrentEntryChange(
```

Comment:
```suggestion
  onBrowserCurrentEntryChange(
```
Just thinking out loud here... `nonRouter` is pretty ambiguously "any change that's not the router" but this is really only going to be the browser, right?

Reply:
I think my plan is to actually only intercept navigations that we know come from the router. So if there's a programmatic handler, we would notify about that as "nonRouter".

Reply:
Okay, I think I understand where you're going with this. When navigate events are intercepted, they end up emitting a `popstate` event even if the source came from `history.pushState`. However, this would be a "router" entry change if it happened during the navigation at a time we expect so we wouldn't want to emit that event to the listener. Things are fine as you have them for now, but when we do at that extra bit of logic, let's update the documentation on this function to explain that behavior.